lib/formatter/index.js

```diff
@@ -86,4 +87,31 @@ Formatter.prototype.unformat = function (options) {
   };
 };
 
+Formatter.prototype.simulateDeletion = function (options) {
+  var deletionStart, deletionEnd;
+  var state = this.unformat.apply(this, arguments);
+  var value = state.value;
+  var selection = state.selection;
+  var delta = Math.abs(state.selection.end - state.selection.start);
+
+  if (delta) {
+    deletionStart = selection.start;
+    deletionEnd = selection.end;
+  } else if (isBackspace(options.event)) {
+    deletionStart = Math.max(0, selection.start - 1);
+    deletionEnd = selection.start;
+  } else { // Handle forward delete
+    deletionStart = selection.start;
+    deletionEnd = Math.min(value.length, selection.start + 1);
+  }
+
+  return {
+    selection: {
+      start: deletionStart,
+      end: deletionStart
```

Comment:
Just checked, Chrome at least doesn't care which direction you start your selection from. The end value will never be less than the start value.